src/probe_matcher.cpp

```diff
@@ -69,7 +69,22 @@ std::set<std::string> ProbeMatcher::get_matches_in_stream(
           char* demangled_name = cxxdemangle(fun_line.c_str());
           if (demangled_name)
           {
-            if (!wildcard_match(prefix + demangled_name, tokens, true, true))
+            // Since we're adding a trailing wildcard, we also have to add an
+            // explicit '(' to the end of the last token (if the user hasn't
+            // specified it already himself) to avoid false matches (e.g. "fun"
+            // matching "function").
+            auto cpp_tokens = tokens;
+            if (!end_wildcard && std::all_of(cpp_tokens.begin(),
+                                             cpp_tokens.end(),
+                                             [&](const std::string& token) {
+                                               return token.find("(") ==
+                                                      std::string::npos;
+                                             }))
+            {
+              cpp_tokens.back() += "(";
+            }
+            if (!wildcard_match(
+                    prefix + demangled_name, cpp_tokens, true, true))
```

Comment:
Feels kinda odd to add a `(` to the end of the search string. Why not remove the `(args...)` from the symbol instead?

Reply:
Good idea, it's even better as it allows to get rid of the artificially added leading and trailing whitespace which may potentially lead to incorrect matches.